src/prototype_room_utils.js

```diff
@@ -1,5 +1,44 @@
 'use strict';
 
+Room.prototype.sortMyRoomsByLinearDistance = function(target) {
+  let sortByLinearDistance = function(object) {
+    return Game.map.getRoomLinearDistance(target, object);
+  };
+
+  return _.sortBy(Memory.myRooms, sortByLinearDistance);
+};
+
+Room.prototype.closestSpawn = function(target) {
+  let pathLength = {};
+  let roomsMy = this.sortMyRoomsByLinearDistance(target);
+
+  for (let room of roomsMy) {
+    let route = Game.map.findRoute(room, target);
```

Comment:
I think we should avoid occupied rooms, like this: https://github.com/TooAngel/screeps/blob/master/src/prototype_creep_routing.js#L40